### PR TITLE
[CDAP-17028] Fixes UI to not show macro functions that doesn't need user input while running preview

### PR DESCRIPTION
**Pull request by ajainarayanan**
Cherry-pick of https://github.com/cdapio/cdap/pull/13160

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js [hunk 1]
```diff
@@ -381,11 +381,21 @@ class HydratorPlusPlusTopPanelCtrl {
...
+
+            This is a temporary fix to not surface simple cases of macro functions.
+            Hence the specific check for known macro functions that doesn't need user input.
           */
           if (macroString &&
             typeof macroString === 'string' &&
             macroString.indexOf('${') !== -1 &&
-            macroString.indexOf('}') !== -1
+            macroString.indexOf('}') !== -1 &&
+            macroString !== '${logicalStartTime()}' &&
```
Comment (bdmogal) [inline cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js]: logicalstarttime can have arguments right, like format?

File: cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js [hunk 2]
```diff
@@ -381,11 +381,21 @@ class HydratorPlusPlusTopPanelCtrl {
...
+
+            This is a temporary fix to not surface simple cases of macro functions.
+            Hence the specific check for known macro functions that doesn't need user input.
           */
           if (macroString &&
             typeof macroString === 'string' &&
             macroString.indexOf('${') !== -1 &&
-            macroString.indexOf('}') !== -1
+            macroString.indexOf('}') !== -1 &&
+            macroString !== '${logicalStartTime(}' &&
```
Comment (njbriggs) [inline cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js]: Is this correct?

Comment (njbriggs) [reply cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js]: As in, should this be `!==` or `!macroString.startsWith`?

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js]: Have fixed it..

**Review by elfenheart — APPROVED** [review]

File: cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js [hunk 3]
```diff
@@ -381,11 +381,21 @@ class HydratorPlusPlusTopPanelCtrl {
...
+            Hence the specific check for known macro functions that doesn't need user input.
           */
           if (macroString &&
             typeof macroString === 'string' &&
             macroString.indexOf('${') !== -1 &&
-            macroString.indexOf('}') !== -1
+            macroString.indexOf('}') !== -1 &&
+            macroString.indexOf('${logicalStartTime(') !== -1 &&
+            macroString.indexOf('${secure(') === -1 &&
+            macroString.indexOf('${oauth(') === -1
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js:398]: these should be `!==` as well?

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/create/toppanel-ctrl.js:398]: Actually this should be the other way. We shouldn't parse for macros if the property is set with one of the macro functions. Have updated

**Review by elfenheart — APPROVED** [review]

**Pull request merged**